src/main/java/org/openjdk/jextract/clang/Cursor.java

```diff
@@ -237,10 +237,10 @@ public boolean equals(Object other) {
         if (this == other) {
             return true;
         }
-        if (!(other instanceof Cursor)) {
+        if (!(other instanceof Cursor otherCursor)) {
             return false;
         }
-        return (Index_h.clang_equalCursors(cursor, ((Cursor)other).cursor) != 0);
+        return (Index_h.clang_equalCursors(cursor, otherCursor.cursor) != 0);
```

Comment:
I think in these cases we can do better by merging the `instanceof` check above this in the condition. i.e.
```suggestion
        return other instanceof Cursor otherCursor
            && (Index_h.clang_equalCursors(cursor, otherCursor.cursor) != 0);
```

And similar for other places that are computing a boolean value, such as all the equals methods.

Reply:
Strictly speaking, the whole method body could be merged into one expression. Its the realm of infinite bikesheds though, and I did not / do not know if there is a set of implicit style choices there.

Reply:
Well, we're on the topic of style changes already. This is new territory, so I think this is where we decide this :)

Merging the `instanceof` into the expression avoids having the pattern variable be carried over to outside of the `if` block, which I think is better.

The `this == other` check above that doesn't have the same problem.

Reply:
I don't have any particularly strong opinions there, so I made that as the change, leaving the `this == other` checks alone